### PR TITLE
Fix component version when download from source

### PR DESCRIPTION
**Pull request by anitacaron**
Related to https://github.com/obophenotype/uberon/pull/2975

### CODE REVIEW

**Comment by anitacaron** [comment]
I tested on the linked PR and I'm getting the error:

```
if [ true  = true ] && [ true  = true ]; then robot --catalog catalog-v001.xml merge -I https://raw.githubusercontent.com/hubmapconsortium/ccf-validation-tools/master/owl/UB_ASCTB_subset.owl \
annotate --ontology-iri http://purl.obolibrary.org/obo/uberon/component-download-hra_subset.owl annotate -V http://purl.obolibrary.org/obo/uberon/releases/2023-09-27/component-download-hra_subset.owl --annotation owl:versionInfo 2023-09-27 -o tmp/component-download-hra_subset.owl.owl; fi
if [ true  = true ]; then if cmp -s tmp/component-download-hra_subset.owl.owl components/hra_subset.owl ; then echo "Component identical."; \
      else echo "Component is different, updating." && robot --catalog catalog-v001.xml annotate -i component-download-hra_subset.owl --ontology-iri http://purl.obolibrary.org/obo/uberon/components/hra_subset.owl annotate -V http://purl.obolibrary.org/obo/uberon/releases/2023-09-27/components/hra_subset.owl --annotation owl:versionInfo 2023-09-27 -o components/hra_subset.owl; fi; fi
Component is different, updating.
org.semanticweb.owlapi.io.OWLOntologyInputSourceException: java.io.FileNotFoundException: component-download-hra_subset.owl (No such file or directory)
Use the -vvv option to show the stack trace.
Use the --help option to see usage information.
make: *** [Makefile:417: components/hra_subset.owl] Error 1
```

The output file for the rule `component-download-hra_subset.owl` is `tmp/component-download-hra_subset.owl.owl` and it's the pre-requisit in the rule `components/hra-subset.owl`. Is there a reason to not directly name the rule `tmp/component-download-hra_subset.owl`?

**Comment by anitacaron** [comment]
The component test in the QC here is failing for the same reason.

**Comment by matentzn** [comment]
The main point was to make the rule .PHONY, so that its is _always run_. If its not .PHONY, you wont run this rule if none of the dependencies have changed, which can result in the weird situation that mirrors or component sources are never updated until someone explicitly runs with -B.

**Comment by anitacaron** [comment]
Sorry, the QC failed because I used the wrong input file in the robot command. I fixed that, and everything is good now.

File: template/src/ontology/Makefile.jinja2 [hunk 1]
```diff
@@ -550,8 +550,9 @@ component-download-{{ component.filename }}: | $(TMPDIR)
 	remove {% if component.base_iris is not none %}{% for iri in component.base_iris %}--base-iri {{iri}} {% endfor %}{% else %}--base-iri $(OBOBASE)/{{ project.id.upper() }} {% endif %}--axioms external --preserve-structure false --trim false \{% endif %}
 	annotate --ontology-iri $(ONTBASE)/$@ $(ANNOTATE_ONTOLOGY_VERSION) -o $(TMPDIR)/$@.owl; fi
 
-$(COMPONENTSDIR)/{{ component.filename }}:  component-download-{{ component.filename }}
-	if [ $(COMP) = true ] ; then if cmp -s $(TMPDIR)/component-download-{{ component.filename }}.owl $@ ; then echo "Component identical."; else echo "Component is different, updating." && cp $(TMPDIR)/component-download-{{ component.filename }}.owl $@; fi; fi
+$(COMPONENTSDIR)/{{ component.filename }}: component-download-{{ component.filename }}
+	if [ $(COMP) = true ]; then if cmp -s $(TMPDIR)/component-download-{{ component.filename }}.owl $@ ; then echo "Component identical."; \
```
Comment (matentzn) [inline template/src/ontology/Makefile.jinja2:554]: Ensure that the cmp Check ever returns "identical"

Comment (anitacaron) [reply template/src/ontology/Makefile.jinja2]: yeah, it wasn't ever identical. I copied the file to another tmp file and then compared the component with that. Now it's good. Ready for review.

**Review by matentzn — DISMISSED** [review]
I think this looks great, not very elegant, but very practical, and we will probably re-use this pattern in other cases as well. I would kinda like if you could get Damien to take a look, just because this is important.

File: template/src/ontology/Makefile.jinja2 [hunk 2]
```diff
@@ -550,8 +550,9 @@ component-download-{{ component.filename }}: | $(TMPDIR)
 	remove {% if component.base_iris is not none %}{% for iri in component.base_iris %}--base-iri {{iri}} {% endfor %}{% else %}--base-iri $(OBOBASE)/{{ project.id.upper() }} {% endif %}--axioms external --preserve-structure false --trim false \{% endif %}
 	annotate --ontology-iri $(ONTBASE)/$@ $(ANNOTATE_ONTOLOGY_VERSION) -o $(TMPDIR)/$@.owl; fi
 
-$(COMPONENTSDIR)/{{ component.filename }}:  component-download-{{ component.filename }}
-	if [ $(COMP) = true ] ; then if cmp -s $(TMPDIR)/component-download-{{ component.filename }}.owl $@ ; then echo "Component identical."; else echo "Component is different, updating." && cp $(TMPDIR)/component-download-{{ component.filename }}.owl $@; fi; fi
+$(COMPONENTSDIR)/{{ component.filename }}: component-download-{{ component.filename }}
+	if [ $(COMP) = true ]; then if cmp -s $(TMPDIR)/component-download-{{ component.filename }}.owl $(TMPDIR)/component-download-{{ component.filename }}.tmp.owl ; then echo "Component identical."; \
```
Comment (gouttegd) [inline template/src/ontology/Makefile.jinja2:554]: What would happen on a “clean” repo if there is no `$(TMPDIR)/component-download- {{ component.filename }}.tmp.owl` file from a previous run?

Comment (matentzn) [reply template/src/ontology/Makefile.jinja2:554]: Good point.. Maybe `if not exist touch`?

Comment (gouttegd) [reply template/src/ontology/Makefile.jinja2:554]: Either that or something like “if the file does not exist, then go to the ‘not-identical’ branch”.

Comment (anitacaron) [reply template/src/ontology/Makefile.jinja2:559]: I tested in an uberon branch without suggested changes. I removed all files `tmp/component-download-*` and it worked as expected, "go to the 'not-identical' branch".

```bash
if [ true  = true ] && [ true  = true ]; then robot --catalog catalog-v001.xml merge -I https://raw.githubusercontent.com/hubmapconsortium/ccf-validation-tools/master/owl/UB_ASCTB_subset.owl \
annotate --ontology-iri http://purl.obolibrary.org/obo/uberon/component-download-hra_subset.owl annotate -V http://purl.obolibrary.org/obo/uberon/releases/2024-01-03/component-download-hra_subset.owl --annotation owl:versionInfo 2024-01-03 -o tmp/component-download-hra_subset.owl.owl; fi
if [ true  = true ]; then if cmp -s tmp/component-download-hra_subset.owl.owl tmp/component-download-hra_subset.owl.tmp.owl ; then echo "Component identical."; \
      else echo "Component is different, updating." && cp tmp/component-download-hra_subset.owl.owl tmp/component-download-hra_subset.owl.tmp.owl && robot --catalog catalog-v001.xml annotate -i tmp/component-download-hra_subset.owl.owl --ontology-iri http://purl.obolibrary.org/obo/uberon/components/hra_subset.owl annotate -V http://purl.obolibrary.org/obo/uberon/releases/2024-01-03/components/hra_subset.owl --annotation owl:versionInfo 2024-01-03 -o components/hra_subset.owl; fi; fi
Component is different, updating.
```

Comment (matentzn) [reply template/src/ontology/Makefile.jinja2:559]: Yep, you are right @anitacaron:

```
➜  ontology git:(fix-abbrev) ✗ cmp -s gorgon.owl hp-edit.owl || echo "FAIL"
FAIL
➜  ontology git:(fix-abbrev) ✗ cmp -s hp-edit.owl hp-edit.owl || echo "FAIL" 
➜  ontology git:(fix-abbrev) ✗ 
```

Seems when the file does not exist, cmp evaluates to false, which is the desired behaviour.

I think we can close this comment.

**Review by matentzn — APPROVED** [review]

**Pull request merged**